specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2024-06-01-preview/nodetype.json

```diff
@@ -1528,6 +1538,48 @@
       },
       "description": "Specifies set of certificates that should be installed onto the virtual machines."
     },
+    "VmApplication": {
+      "type": "object",
+      "required": [
+        "packageReferenceId"
+      ],
+      "properties": {
+        "configurationReference": {
+          "type": "string",
+          "description": "Optional, Specifies the uri to an azure blob that will replace the default configuration for the package if provided."
+        },
+        "enableAutomaticUpgrade": {
+          "type": "boolean",
```

Comment:
Similar to earlier comment: please use enums instead of booleans. In this case, would something like upgradePolicy: { Automatic, Manual } work?